homepageapp/templates/registration/register.html

```diff
@@ -0,0 +1,40 @@
+{% extends 'homepageapp/base.html' %}
+
+{% block content %}
+
+    {% for message in messages %}
+    {% csrf_token %}
+        <strong>Message: </strong> {{ message }}
+    {% endfor %}
+
+    <h3>Register form:</h3>
+
+    <form method="post">
+        {% csrf_token %}
+
```

Comment:
Try simply 
`{{form.as_p}}`
it will generate the form for you and also handle error messages. 
Issue here is there will be no errors shown if you, for example, put really short password.